pywt/_extensions/_dwt.pyx

```diff
@@ -98,9 +98,9 @@ cpdef dwt_axis(np.ndarray data, Wavelet wavelet, MODE mode, unsigned int axis=0)
     return (cA, cD)
 
 
-cpdef idwt_single(data_t[::1] cA, data_t[::1] cD, Wavelet wavelet, MODE mode):
-    cdef data_t[::1] rec
+cpdef idwt_single(np.ndarray cA, np.ndarray cD, Wavelet wavelet, MODE mode):
```

Comment:
I have Cython 0.23 installed, which is why I am confused that the following works:

```
>>> import pywt, numpy as np
>>> w = pywt.Wavelet('haar')
>>> m = pywt.Modes.zero
>>> a = np.arange(10, dtype='double') ** 2
>>> cA, cD = pywt.dwt(a, w, m)
>>> pywt._extensions._dwt.idwt_single(cA.astype('float'), cD.astype('double'), w, m)
<MemoryView of 'ndarray' at 0x7f92de51c398>
>>> np.asarray(_)
array([  0.,   1.,   4.,   9.,  16.,  25.,  36.,  49.,  64.,  81.])
>>> a
array([  0.,   1.,   4.,   9.,  16.,  25.,  36.,  49.,  64.,  81.])
```

My impression was that it shouldn't, and if something was going wrong (i.e. a buffer of floats interpreted as doubles) the result should be incorrect.